tests/api/test_pagination.py

```diff
@@ -0,0 +1,1654 @@
+    content = get_graphql_content(response)
+    page_info = content["data"][resolver_name]["pageInfo"]
+
+    assert not page_info["hasPreviousPage"]
+    assert page_info["hasNextPage"]
+    assert page_info["startCursor"] == start_cursor_page1
+    assert page_info["endCursor"] == end_cursor_page1
+
+    variables = {"first": 3, "after": end_cursor_page1, "sortBy": order_by}
+    response = staff_api_client.post_graphql(query, variables)
+    content = get_graphql_content(response)
+    page_info = content["data"][resolver_name]["pageInfo"]
+
+    assert page_info["hasPreviousPage"]
+    assert not page_info["hasNextPage"]
+    assert page_info["startCursor"] == start_cursor_page2
+    assert page_info["endCursor"] == end_cursor_page2
+
+    variables = {"last": 2, "before": start_cursor_page2, "sortBy": order_by}
+    response = staff_api_client.post_graphql(query, variables)
+    content = get_graphql_content(response)
+    page_info = content["data"][resolver_name]["pageInfo"]
+
+    assert page_info["hasPreviousPage"]
+    assert page_info["hasNextPage"]
+    assert page_info["startCursor"] == start_cursor_before
+    assert page_info["endCursor"] == end_cursor_page1
+
+
+@pytest.mark.parametrize(
+    "order_by, cursor_fields, user_order",
+    [
+        (
+            None,
+            ["email"],
+            [
+                "allen@example.com",
+                "ddevito@example.com",
+                "ellen@example.com",
+                "leslie@example.com",
+                "zordon@example.com",
+            ],
+        ),
+        (
+            {"field": "FIRST_NAME", "direction": "ASC"},
+            UserSortField.FIRST_NAME.value,
+            [
+                "zordon@example.com",
+                "leslie@example.com",
+                "ddevito@example.com",
+                "ellen@example.com",
+                "allen@example.com",
+            ],
+        ),
+        (
+            {"field": "LAST_NAME", "direction": "ASC"},
+            UserSortField.LAST_NAME.value,
+            [
+                "allen@example.com",
+                "ellen@example.com",
+                "ddevito@example.com",
+                "zordon@example.com",
+                "leslie@example.com",
+            ],
+        ),
+        (
+            {"field": "EMAIL", "direction": "ASC"},
+            UserSortField.EMAIL.value,
+            [
+                "allen@example.com",
+                "ddevito@example.com",
+                "ellen@example.com",
+                "leslie@example.com",
+                "zordon@example.com",
+            ],
+        ),
+        (
+            {"field": "EMAIL", "direction": "DESC"},
+            UserSortField.EMAIL.value,
+            [
+                "zordon@example.com",
+                "leslie@example.com",
+                "ellen@example.com",
+                "ddevito@example.com",
+                "allen@example.com",
+            ],
+        ),
+    ],
+)
+def test_user_pagination(
```

Comment:
I can't tell what this test is checking as all of the data is prepared outside of test's body, all assertions are made outside of test's body and expected output refers to a field different than the one used for sorting so even verifying that the assumptions are correct is impossible. It seems to be five separate tests disguised as one.

It would be easier to read if it was five separate tests with fields used for sorting explicitly being created (or updated) inside the test body finished with an assertion that a particular field is properly sorted (eg. `assert [user["foo"] for user in results] == ["a", "b", "c", "d", "e"]`).